cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -505,26 +532,43 @@ CiderRuntimeModule::fetchResults(int32_t max_row) {
       ciderCompilationResult_->impl_->query_mem_desc_->getQueryDescriptionType();
   bool is_project = query_mem_desc_t == QueryDescriptionType::Projection;
   if (is_project) {
-    auto schema = ciderCompilationResult_->getOutputCiderTableSchema();
-    int column_num = schema.getColumnCount();
+    if (ciderCompilationOption_.use_cider_data_format) {
+      std::function<void(CiderBatch*)> process_function =
+          [row_num = total_matched_rows_, &process_function](CiderBatch* batch) -> void {
+        batch->setLength(row_num);
+        batch->setNullCount(row_num -
+                            CiderBitUtils::countSetBits(batch->getNulls(), row_num));
+        for (size_t i = 0; i < batch->getChildrenNum(); ++i) {
+          auto child = batch->getChildAt(i);
+          process_function(child.get());
+        }
```

Comment:
why not define a common utils function?

Reply:
There is no other place needs, seems unnecessary now.